### PR TITLE
Handled ignorable files

### PR DESCRIPTION
**Pull request by PA-Rweins**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/43694)

## Description
Some files that were supposed to be ignored were not ignored. 
In order to ignore them I created a list of those files and created a function where I check for ignorable files so they wont be validated.

## Must have
- [x] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

**Review by Bargenish — CHANGES_REQUESTED** [review]
Let's discuss the tests offline.

**Comment by tomneeman151293** [comment]
@PA-Rweins Please add a test before your changes, showing that the test fails with same error msg in bug, and after that see that test passes after your changes.

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 1]
```diff
@@ -1124,31 +1124,33 @@ def test_ignore_test_doc_non_pack_file_should_not_ignore(self, file_path: str):
...
             - File is ignored and True is returned
         """
         validate_manager = ValidateManager(check_is_unskipped=False)
-        assert validate_manager.ignore_test_doc_non_pack_file(file_path)
+        assert validate_manager.ignore_files_irrelevant_for_validation(file_path)
 
     @pytest.mark.parametrize('file_path',
                              ['OtherDir/Integration/file.json',
                               'TestData/file.json',
                               'TestPlaybooks/file.yml',
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/tests/validators_test.py:1151]: in general, this folder shouldn't be ignored but let's leave it as is (better to deprecate the tests in it).

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -1404,9 +1404,10 @@ def format_file_path(self, file_path, old_path, old_format_files):
         else:
             return file_path
 
-    def ignore_test_doc_non_pack_file(self, file_path: str) -> bool:
+    def ignore_files_irrelevant_for_validation(self, file_path: str) -> bool:
         # ignore doc data, test_data and non pack files
-        if any(test_dir in str(file_path) for test_dir in TESTS_AND_DOC_DIRECTORIES) or PACKS_DIR not in file_path:
+        if any(name in str(file_path) for name in VALIDATION_USING_GIT_IGNORABLE_DATA) \
+                or PACKS_DIR not in file_path:
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/validate_manager.py:1410]: I would replace the order of the conditions (it is better performance-wise)

Comment (PA-Rweins) [reply demisto_sdk/commands/validate/validate_manager.py]: Fixed

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 1 above)
Comment (PA-Rweins) [reply demisto_sdk/commands/validate/tests/validators_test.py:1153]: I'm not sure which folder you're talking about

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -1392,7 +1392,7 @@ def format_file_path(self, file_path, old_path, old_format_files):
                 file_path = file_path.replace('.py', '.yml').replace('.ps1', '.yml').replace('.js', '.yml')
 
                 if old_path:
-                    old_path = old_path.replace('.py', '.yml').replace('.ps1', ',yml').replace('.js', '.yml')
+                    old_path = old_path.replace('.py', '.yml').replace('.ps1', '.yml').replace('.js', '.yml')
```
Comment (Bargenish) [inline demisto_sdk/commands/validate/validate_manager.py:1395]: Good catch!

File: demisto_sdk/commands/common/constants.py [hunk 4]
```diff
@@ -858,6 +858,19 @@ class FileType(Enum):
     'doc_imgs',
 ]
 
+VALIDATION_USING_GIT_IGNORABLE_DATA = (
+    'Pipfile',
+    'Pipfile.lock',
+    'command_examples',
```
Comment (Bargenish) [inline demisto_sdk/commands/common/constants.py:864]: Im pretty worried about  this one. we do not enforce the name of the example file and we have various names on the repo.
see the following for examples:
Packs/SentinelOne/Integrations/SentinelOne-V2/examples.txt
Packs/BitSight/Integrations/BitSightForSecurityPerformanceManagement/command-examples.txt

**Review by Bargenish — CHANGES_REQUESTED** [review]
great job with the testing

**Review by Bargenish — APPROVED** [review]

**Review by yaakovi — APPROVED** [review]

**Pull request merged**